frontend/test/metabase/scenarios/question/custom_column.cy.spec.js

```diff
@@ -414,6 +415,28 @@ describe("scenarios > question > custom columns", () => {
     cy.contains("37.65");
   });
 
+  describe("data type", () => {
+    it("should understand string functions", () => {
```

Comment:
I haven't followed that type refactor, but are there any other types that are inferred?

Reply:
Yes, a good starting point is to follow the plan I list in #15952. For instance, the unit tests in #15940 demonstrates the existing coverage.